### PR TITLE
Docs/server static pages

### PR DESCRIPTION
**Pull request by noctera**
This PR adds the documentation for the new "static pages" update in vocascan/vocascan-server#72

To Do:
- [x] config example for env file

### CODE REVIEW

File: docs/vocascan-server/configuration.md [hunk 1]
```diff
@@ -107,6 +107,16 @@ For more information about the logger section see [logging guide](vocascan-serve
 | `max_files`    |         | Limit the number of files created when the size of the log file is exceeded.                                                           |
 | `archive_logs` |         | If true, all log files, but the current one will be packed into a zip archive.                                                         |
 
+## Custom Pages (`pages.`)
+
+| Name       | Default | Required | Description                                                                                                                                                                                                                                             |
```
Comment (luwol03) [inline docs/vocascan-server/configuration.md]: please use the table headlines from the other tables

File: docs/vocascan-server/configuration.md [hunk 2]
```diff
@@ -107,6 +107,16 @@ For more information about the logger section see [logging guide](vocascan-serve
 | `max_files`    |         | Limit the number of files created when the size of the log file is exceeded.                                                           |
 | `archive_logs` |         | If true, all log files, but the current one will be packed into a zip archive.                                                         |
 
+## Custom Pages (`pages.`)
```
Comment (luwol03) [inline docs/vocascan-server/configuration.md]:
```suggestion
## Custom Pages (`pages.*`)
```

File: docs/vocascan-server/configuration.md [hunk 3]
```diff
@@ -107,6 +107,16 @@ For more information about the logger section see [logging guide](vocascan-serve
...
+
+| Name       | Default | Required | Description                                                                                                                                                                                                                                             |
+| ---------- | ------- | -------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `url`      |         | `true`   | The route on which your page will be displayed                                                                                                                                                                                                          |
```
Comment (luwol03) [inline docs/vocascan-server/configuration.md]: What about adding relative to `/`?

File: docs/vocascan-server/configuration.md [hunk 4]
```diff
@@ -107,6 +107,16 @@ For more information about the logger section see [logging guide](vocascan-serve
...
+| `fallback` |         | `true`   | Defines a fallback page, if neither a language is specified per query parameter (`?lang=`), nor the queried language exists. You can either set a static page or a redirect to one of the defined languages (recommended if you use multiple languages) |
+| `type`     |         | `true`   | Defines whether the server renders a static page or just redirects to another. Options: `file` or `redirect`                                                                                                                                            |
+| `location` |         | `true`   | Either the path to your html file, or an internet address for the redirect                                                                                                                                                                              |
+| `langs`    |         | `false`  | Option to define multiple language for your static file or redirect                                                                                                                                                                                     |
```
Comment (luwol03) [inline docs/vocascan-server/configuration.md]: please define that a better how the langs are structured.

File: docs/vocascan-server/customPages.md [hunk 5]
```diff
@@ -0,0 +1,131 @@
+# Custom Pages
+
+We provide you with the possibility to host your own static pages or make redirects to other websites on your Vocascan
+server and make them accessible to the public. This can be very helpful if you have to comply with legal requirements
+and therefore have to provide a privacy policy or other information. To add your page to the server, you need to add the
+files as a volume to the Docker container and define them in `vocascan.config.js` or the `env variables`.
```
Comment (luwol03) [inline docs/vocascan-server/customPages.md]: Do that guide a more generally. So also user using other environments like webspaces can understand the guide. You can then specify it how it works in docker.

File: docs/vocascan-server/customPages.md [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+
+!> The Vocascan frontend needs predefined routes (`url`) to find specific pages:</br> 1. Legal Notice:
```
Comment (luwol03) [inline docs/vocascan-server/customPages.md]: what about adding that as a table so if we later add some more?

File: docs/vocascan-server/customPages.md [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+
+Open your `docker-compose.yml` file
+
+```bash
+nano docker-compose.yml
+```
+
+and add the volume for your `staticPages` folder
+
+```yaml
```
Comment (luwol03) [inline docs/vocascan-server/customPages.md]: do we use yaml or yml?

File: docs/vocascan-server/installation/traefik.md [hunk 8]
```diff
@@ -135,10 +135,12 @@ On top of that, replace this parts to your domain
                               ...
 18   - "traefik.frontend.rule=Host:monitor.your_domain.com"
                               ...
-33   - "traefik.http.routers.vocascan.rule=Host(`web.your_domain.com`)"
+33   - "traefik.http.routers.vocascan.rule=Host(`your_domain.vocascan.com`) && PathPrefix(`/p/`, `/api`)"
```
Comment (luwol03) [inline docs/vocascan-server/installation/traefik.md]: remove vocascan

**Review by luwol03 — CHANGES_REQUESTED** [review]

File: docs/vocascan-server/customPages.md (hunk 6 above)
Comment (noctera) [reply docs/vocascan-server/customPages.md]: I think you can't add a table to a `!>` tag

File: docs/vocascan-server/customPages.md (hunk 7 above)
Comment (noctera) [reply docs/vocascan-server/customPages.md]: yml. I made a mistake there

File: docs/vocascan-server/customPages.md (hunk 6 above)
Comment (luwol03) [reply docs/vocascan-server/customPages.md]: I think that's not possible, but there is no need for a !> block. Maybe just use a separate headline where you describe for what these paths are.

File: docs/vocascan-server/customPages.md [hunk 9]
```diff
@@ -101,6 +105,25 @@ pages: {
 ...
 ```
 
+### **Environment variables**
+
+An ENV config example could look like this.
+
+```env
```
Comment (luwol03) [inline docs/vocascan-server/customPages.md:119]: I dont think that this example is necessary :D - but if you already have it, you can may add a details tag around it, so it doesnt fill the page that much.

Comment (noctera) [reply docs/vocascan-server/customPages.md:119]: This is already in a tabs section. You can decide whether you want to see the config file or the env options

Comment (luwol03) [reply docs/vocascan-server/customPages.md:119]: ok. then its ok.

File: docs/vocascan-server/configuration.md [hunk 10]
```diff
@@ -107,6 +107,16 @@ For more information about the logger section see [logging guide](vocascan-serve
...
+| `fallback` |         | Defines a fallback page, if neither a language is specified per query parameter (`?lang=`), nor the queried language exists. You can either set a static page or a redirect to one of the defined languages (recommended if you use multiple languages) |
+| `type`     |         | Defines whether the server renders a static page or just redirects to another. Options: `file` or `redirect`                                                                                                                                            |
+| `location` |         | Either the path to your html file, or an internet address for the redirect                                                                                                                                                                              |
+| `langs`    |         | Option to define multiple languages for your static file or redirect. Add props, named after the country code (ISO-639-1: Germany -> de), which include a `type` and a `location`                                                                       |
```
Comment (luwol03) [inline docs/vocascan-server/configuration.md]: can you extract type and location into two different rows. Maybe `(fallback|langs).type` or even better, only `lang.type` and then mention in fallback, that the props from langs are valid. And additionally I would suggest to add a link to that ISO norm.

File: docs/vocascan-server/customPages.md [hunk 11]
```diff
@@ -5,8 +5,15 @@ server and make them accessible to the public. This can be very helpful if you h
...
 
-!> The Vocascan frontend needs predefined routes (`url`) to find specific pages:</br> 1. Legal Notice:
-`/p/legal-notice`</br>2. Privacy Policy: `/p/privacy-policy`</br>3. Terms and Conditions: `/p/terms-and-conditions`
+## Predefined routes
+
+The Vocascan frontend needs predefined routes (`url`) to find specific pages:
+
+| Type                 | Route                   |
+| -------------------- | ----------------------- |
+| Legal Notice         | /p/legal-notice         |
```
Comment (luwol03) [inline docs/vocascan-server/customPages.md]: can you add the route into inline code blocks

File: docs/vocascan-server/configuration.md [hunk 12]
```diff
@@ -109,13 +109,15 @@ For more information about the logger section see [logging guide](vocascan-serve
...
+| `fallback.type`           |         | Defines whether the server renders a static page or just redirects to another. Options: `file` or `redirect`                                                                                                                                            |
+| `fallback.location`       |         | Either the path to your html file, or an internet address for the redirect                                                                                                                                                                              |
+| `langs`                   |         | Option to define multiple languages for your static file or redirect. Add props, named after the country code ([ISO-639-1](https://en.wikipedia.org/wiki/List_of_ISO_639-1_codes): Germany -> de)                                                       |
+| `langs.ISO_LANG.type`     |         | Defines whether the server renders a static page or just redirects to another. Options: `file` or `redirect`                                                                                                                                            |
```
Comment (luwol03) [inline docs/vocascan-server/configuration.md]: I suggest adding something like this to indicate that ISO_LANG is a placeholder. You can decide if you want to use <> or [].
```suggestion
| `langs.<ISO_LANG>.type`     |         | Defines whether the server renders a static page or just redirects to another. Options: `file` or `redirect`                                                                                                                                            |
```

File: docs/vocascan-server/customPages.md [hunk 13]
```diff
@@ -0,0 +1,161 @@
...
+```
+
+and add your static file and redirects
+
+?> In our example, we created the volume for our files under `/root/vocascan/`. This is why we use this path as our
+absolute path. If you don't use Docker or used a different path, please make sure to take this.
+
+```js
+...
+pages: {
```
Comment (luwol03) [inline docs/vocascan-server/customPages.md]: fix the indentation
```suggestion
  pages: {
```

File: docs/vocascan-server/configuration.md [hunk 14]
```diff
@@ -109,15 +109,15 @@ For more information about the logger section see [logging guide](vocascan-serve
...
+| `fallback.type`             |         | Defines whether the server renders a static page or just redirects to another. Options: `file` or `redirect`                                                                                                                                            |
+| `fallback.location`         |         | Either the path to your html file, or an internet address for the redirect                                                                                                                                                                              |
+| `langs`                     |         | Option to define multiple languages for your static file or redirect. Add props, named after the country code ([ISO-639-1](https://en.wikipedia.org/wiki/List_of_ISO_639-1_codes): Germany -> de)                                                       |
+| `langs.ISO_LANG.type`       |         | Defines whether the server renders a static page or just redirects to another. Options: `file` or `redirect`                                                                                                                                            |
```
Comment (luwol03) [inline docs/vocascan-server/configuration.md]: also here

**Review by luwol03 — APPROVED** [review]

**Pull request merged**